packages/contracts-periphery/deploy/op-nft/AttestationStationProxy.ts

```diff
@@ -0,0 +1,32 @@
+/* Imports: External */
+import { DeployFunction } from 'hardhat-deploy/dist/types'
+import '@eth-optimism/hardhat-deploy-config'
+import '@nomiclabs/hardhat-ethers'
+import 'hardhat-deploy'
+
+const deployFn: DeployFunction = async (hre) => {
+  const { deployer } = await hre.getNamedAccounts()
+  const { deploy } = hre.deployments
+
+  console.log(`Deploying AttestationStationProxy to ${hre.network.name}`)
+  console.log(`Using deployer ${deployer}`)
+
+  await deploy('AttestationStationProxy', {
+    contract: 'Proxy',
+    from: deployer,
+    args: [deployer],
+    log: true,
+    waitConfirmations: 1,
+  })
+
+  const Deployment__AttestationStationProxy = await hre.deployments.get(
+    'AttestationStationProxy'
+  )
+  console.log(
+    `AttestationStationProxy deployed to ${Deployment__AttestationStationProxy.address}`
+  )
```

Comment:
You should get the implementation on the proxy via `eth_call` on this function:
https://github.com/ethereum-optimism/optimism/blob/d30e727f3116ab1d3e7226289f5238376c4a4fd7/packages/contracts-bedrock/contracts/universal/Proxy.sol#L133

Note that its not a `view` function so using ethers you need to `Contract.callStatic.implementation()` otherwise it will try to send a tx.

If the impl is not set to the AttestationStation then you will need to set the impl to the address of the AttestationStation. You could get the address through a companion network or compute the address client side since its a create2 deployment

You then need to transfer ownership of the proxy to the L1 proxyadmin (cant remember if we have deployed a L1 proxyadmin yet or not), if not then governance multisig. There will be a proxyadmin deploy artifact in the deployments if we did deploy it

Reply:
The owner of the proxy is the constructor arg, to make the deployment simple you can make the initial admin the deployer (this is what you are doing) then you can set the impl (prevents the need to have the multisig overhead) then transfer the ownership away from the deployer key.

Note that you can also `eth_call` the admin value and check to see if its the deployer and if so then you can try to transfer it.

Generally the design pattern we are going for is that you can stop and restart the script without issues